libs/docs/src/docs/2.2.0/innerloop-vs-outerloop.md

```diff
@@ -0,0 +1,420 @@
+        kind: run
+        isDefault: true
+  - id: debug
+    exec:
+      component: runtime
+      commandLine: npm run debug
+      workingDir: ${PROJECT_SOURCE}
+      group:
+        kind: debug
+        isDefault: true
+  - id: test
+    exec:
+      component: runtime
+      commandLine: npm test
+      workingDir: ${PROJECT_SOURCE}
+      group:
+        kind: test
+        isDefault: true
+```
+
+The component and commands here allows the developer to build, run, debug, and test their project within a local cluster using a devfile supported development tool (e.g. `odo`).
+
+## What is outerloop?
+
+**Outerloop** deployments are ones done after the development stage once the source is checked into the source repository. Such deployments would include integration testing, full builds or deployments.
+
+### Procedure
+
+1. Add an image component to define the image building process
+    1. Label your image tag with `imageName`
+    2. Define your `dockerfile` with your file path, build context and if your build requires root privileges
+
+    ```yaml
+    components:
+      - name: outerloop-build
+        image:
+          imageName: landingpage-image:latest
+          dockerfile:
+            uri: docker/Dockerfile
+            buildContext: .
+            rootRequired: false
+    ```
+
+2. Add a command which will instruct your deployment to build the docker image using the `image` component
+
+    ```yaml
+    commands:
+      - id: build-image
+        apply:
+          component: outerloop-build
+    ```
+
+3. Add a deployment component which best suits your cluster’s runtime environment (`kubernetes`/`openshift`), in this case openshift
+    1. Provide the file path to your openshift template file with `uri`
+
+    ```yaml
+    components:
+      - name: outerloop-build
+        image:
+          imageName: landingpage-image:latest
+            dockerfile:
+              uri: docker/Dockerfile
+              buildContext: .
+              rootRequired: false
+      - name: outerloop-deploy
+        openshift:
+          uri: landingpage-template.yaml
+    ```
+
+4. As with the image component, we will need to create a command for the openshift component
+
+    ```yaml
+    commands:
+      - id: build-image
+        apply:
+          component: outerloop-build
+      - id: deploy-openshift
+        apply:
+          component: outerloop-deploy
+    ```
+
+5. To complete the deployment process, combine the two commands into a complete deploy command using composite
+
+    ```yaml
+    commands:
+      - id: build-image
+        apply:
+          component: outerloop-build
+      - id: deploy-openshift
+        apply:
+          component: outerloop-deploy
+      - id: deploy
+        composite:
+          commands:
+            - build-image
+            - deploy-openshift
+          group:
+            kind: deploy
+            isDefault: true
+    ```
+
+```yaml {% title="Final outerloop devfile" filename="devfile.yaml" %}
```

Comment:
add header here too

Reply:
Sounds good, from a quick glance at the raw file I didn't realize that the title field adds a header